fdbclient/SpecialKeySpace.actor.h

```diff
@@ -38,13 +38,55 @@ class SpecialKeyRangeBaseImpl {
 	// Each derived class only needs to implement this simple version of getRange
 	virtual Future<Standalone<RangeResultRef>> getRange(ReadYourWritesTransaction* ryw, KeyRangeRef kr) const = 0;
 
-	explicit SpecialKeyRangeBaseImpl(KeyRangeRef kr) : range(kr) {}
+	explicit SpecialKeyRangeBaseImpl(KeyRangeRef kr, bool async = false) : range(kr), async(async) {}
 	KeyRangeRef getKeyRange() const { return range; }
+	bool isAsync() const { return async; }
```

Comment:
I'd rather make this virtual, just return false here, and then override it in SpecialKeyRangeAsyncImpl. That way users just need to pick which subclass they want and can't mess up the subclass constructor call

Reply:
Yeah, this is much better. Thanks.